dash-renderer/src/APIController.react.js

```diff
@@ -34,6 +47,18 @@ const UnconnectedContainer = props => {
     }
     const renderedTree = useRef(false);
 
+    const propsRef = useRef({});
+    propsRef.current = props;
+
+    const provider = useRef({
+        fn: () => ({
+            _dashprivate_config: propsRef.current.config,
+            _dashprivate_dispatch: propsRef.current.dispatch,
+            _dashprivate_graphs: propsRef.current.graphs,
+            _dashprivate_loadingMap: propsRef.current.loadingMap,
+        }),
+    });
```

Comment:
These props never affect the decision to re-render a TreeContainer. Instead of passing them and running a check with `shouldComponentUpdate`, provide through a context instead. The context value is a function that gives access to the `current` value of a subset of props coming from the store.